test_crates/enum_variant_imports/src/lib.rs

```diff
@@ -0,0 +1,166 @@
+#![no_std]
+#![allow(unused_imports)]
+
+//! This crate exercises enum variant import paths across namespaces, reexports,
+//! doc-hidden/deprecated modifiers, and glob shadowing.
+//! Expected behavior: variants remain importable where allowed, inherit reexport modifiers,
+//! and obey namespace-specific shadowing rules.
+//!
+//! Exported names (paths are relative to the crate root):
+//! - `Base` and its variants via `Base::{Plain, Deprecated, Hidden, DeprecatedHidden}`.
+//! - `Plain`, `Deprecated`, `Hidden`, `DeprecatedHidden` via `pub use Base::*` (glob reexport).
+//! - `RenamedPlain`, `HiddenPlain`, `DeprecatedPlain` via explicit reexports of `Base::Plain`,
+//!   carrying the corresponding rename and doc/deprecation modifiers.
+//! - `namespace`, `namespace::Colors`, `namespace::Colors::{Red, Green, Blue}`,
+//!   and `namespace::Red` via a glob reexport of enum variants. The `namespace::Red` name is
+//!   both a value (variant) and a type (struct), which is allowed across namespaces.
+//!   `namespace::Green` and `namespace::Blue` are not exported because tuple/unit structs
+//!   of the same names shadow the glob-imported variants in the value namespace.
```

Comment:
```suggestion
//!   `namespace::Green` and `namespace::Blue` are the structs, not the variants,
//!   because the variants are shadowed by the local struct definitions, since
//!   the tuple struct's constructor and the unit struct's value are placed 
//!   in the values namespace.
```